File: app/src/main/java/com/returnnotfound/stain/utils/ContextUtils.java
package com.returnnotfound.stain.utils;

import android.app.Activity;
import android.content.Context;

public abstract class ContextUtils {

  public static boolean isValidContext(Context context) {
    // Context null
    if (context == null) {
      return false;
    }
    // Activity is finishing
    return !(context instanceof Activity) || !((Activity) context).isFinishing();
  }
}

File: app/src/main/java/com/returnnotfound/stain/utils/ContextUtils.kt
package com.returnnotfound.stain.utils

import android.app.Activity
import android.content.Context

object ContextUtils {
  fun isValidContext(context: Context?): Boolean {
    // Context null
    return if (context == null) {
      false
    } else context !is Activity || !context.isFinishing
    // Activity is finishing
  }
}